spec/solver_spec.rb

```
@@ -0,0 +1,31 @@
+require './solver'
+
+describe Solver do
+  context 'Test Solver methods:' do
+    before(:each) do
+      @solver = Solver.new
+    end
+
+    it 'Should calculate factorial of an integer number: ' do
+      expect(@solver.factorial(0)).to eq 1
+      expect(@solver.factorial(1)).to eq 1
+      expect(@solver.factorial(2)).to eq 2
+      expect(@solver.factorial(3)).to eq 6
+      expect(@solver.factorial(10)).to eq 3_628_800
+    end
```

- [ ] Kindly add a test for when the number is negative (it should raise an exception).